src/rgw/rgw_sal_motr.cc

```diff
@@ -1923,127 +1922,130 @@ int MotrObject::MotrDeleteOp::delete_obj(const DoutPrefixProvider* dpp, optional
       if(ent.key.instance == "null")
         del_null_ref_key = true;
     }
-  } else {
-    ldpp_dout(dpp, 20) << "delete " << delete_key << " from " << tenant_bkt_name << dendl;
-    // Remove the motr object.
-    // versioning enabled and suspended case.
-    if (info.versioned()) {
-      if (source->have_instance())
+  }
+  // Remove the motr object.
+  // versioning enabled and suspended case.
```

Comment:
We should handle suspended case separately, if no version id is provided then we should always delete the null object.